AUV/auv_viewer_processing/auv_processing.py

```diff
@@ -0,0 +1,725 @@
+    to populate pgsql tables used by the AUV viewer
+    """
+    dive_report_path = os.path.join(os.path.basename(campaign_path), 'all_reports', '%s.pdf' % dive_name)
+    campaign_name    = os.path.basename(campaign_path)
+    for file in os.listdir(os.path.join(campaign_path, dive_name, 'track_files')):
+        if file.endswith('kml'):
+            kml_path = os.path.join(campaign_name, dive_name, 'track_files', file)
+
+    image_path  = os.path.basename(_geotiff_dive_path(os.path.join(campaign_path, dive_name)))
+    dive_number = get_dive_number(dive_name, campaign_path)
+
+    dive_metadata_uuid = str(uuid.uuid4())
+    platform_code      = 'SIRIUS'
+    pattern            = 'Trajectory'
+    abstract           = get_abstract_dive(os.path.join(campaign_path, dive_name))
+
+    lon_seq   = [x['up_right_lon'] for x in geotiff_metadata]
+    lat_seq   = [x['up_right_lat'] for x in geotiff_metadata]
+    depth_seq = [x['depth'] for x in geotiff_metadata]
+
+    geospatial_lat_min      = min(lat_seq)
+    geospatial_lat_max      = max(lat_seq)
+    geospatial_lon_min      = min(lon_seq)
+    geospatial_lon_max      = max(lon_seq)
+    geospatial_vertical_min = min(depth_seq)
+    geospatial_vertical_max = max(depth_seq)
+    distance_covered_in_m   = compute_track_distance(geotiff_metadata)
+
+    time_seq            = [x['time'] for x in geotiff_metadata]
+    time_coverage_start = min([datetime.strptime(time, '%Y%m%dT%H%M%SZ') for time in time_seq]).strftime('%Y%m%dT%H%M%SZ')
+    time_coverage_end   = max([datetime.strptime(time, '%Y%m%dT%H%M%SZ') for time in time_seq]).strftime('%Y%m%dT%H%M%SZ')
+
+    dive_regexp = re.match('r\d+_\d+_(\w+)', dive_name)
+
+    with open(csv_output_path, 'w') as outcsv:
+        outcsv.write('dive_number,dive_name,dive_metadata_uuid,facility_code,campaign_code,'
+                     'dive_code,distance_covered_in_m,number_of_images,image_path,'
+                     'abstract,platform_code,pattern,dive_report_path,kml_path,'
+                     'geospatial_lat_min, geospatial_lon_min,'
+                     'geospatial_lat_max, geospatial_lon_max,'
+                     'geospatial_vertical_min, geospatial_vertical_max,'
+                     'time_coverage_start, time_coverage_end\n')
+
+        outcsv.write('%d,%s,%s,%s,%s,%s,%f,%d,%s,\"%s\",%s,%s,%s,%s,%f,%f,%f,%f,%f,%f,%s,%s\n'
+                        % (dive_number, dive_regexp.group(1).replace('_', ' '),
+                           dive_metadata_uuid, 'AUV', campaign_name, dive_name,
+                           distance_covered_in_m, len(time_seq), image_path, abstract , platform_code,
+                           pattern, dive_report_path, kml_path,
+                           geospatial_lat_min, geospatial_lon_min,
+                           geospatial_lat_max, geospatial_lon_max,
+                           geospatial_vertical_min, geospatial_vertical_max,
+                           time_coverage_start, time_coverage_end))
```

Comment:
If i remember correctly, csv writer would write the columns in an alphabetical order which was not the desired effect. have to look at it again